### ISSUE TITLE
DBeaver 24.0.0+ disconnect doesn't always work

### ISSUE THREAD

**Issue by xak2000** [dbeaver/dbeaver#29999]
### Description

See the Step 11 of "Steps to Reproduce" section for the description of the bug. I don't know how to describe it in isolation of the steps to reproduce.

### DBeaver Version

Community Edition 24.0.4

### Operating System

Windows 11 22H2

### Database and driver

Database Server: MySQL 5.7.44-google-log
Database Driver: MySQL Connector/J mysql-connector-j-8.4.0 (Revision: 1c3f5c149e0bfe31c7fbeb24e2d260cd890972c4)

### Steps to reproduce

1. Start DBeaver.
2. Make sure no Editors opened (middle part of the screen).
3. Connect to a Database server from the Projects view.
4. Double-click on any TABLE in a DB on the connected server, so the TABLE View (Properties tab) will be opened in the middle of the screen. This is the state you should be in:
![image](https://github.com/dbeaver/dbeaver/assets/1576014/316e7f16-c9c2-414d-b4e9-247c57f8c4e2)
5. Disconnect from the Database in the Project View using context menu. The Middle of the screen should still have an opened TAB with the table, but it says "Disconnected" with "Reconnect to the database" in the middle of the screen. 
 ![image](https://github.com/dbeaver/dbeaver/assets/1576014/c2f8e001-47c4-4342-b819-9953c364d3e9)
 ![image](https://github.com/dbeaver/dbeaver/assets/1576014/d5d59835-6971-4075-a1e5-43e50db3e4a1)

6. Close DBeaver.
7. Start DBeaver.
8. The Middle of the screen should have an opened TAB with the table (Properties tab) in a Connected state.
![image](https://github.com/dbeaver/dbeaver/assets/1576014/69898f0b-f4ee-4356-8b02-180440a8ed0b)
9. Find the DB Server in the Projects View again (by expanding <Group_name> -> Connections -> <Folder_name> -> <DB_Connection_name>. The UI shows a green checkbox on the DB Connection, that means it is connected.
![image](https://github.com/dbeaver/dbeaver/assets/1576014/4f1938a8-13d7-4569-848a-6422442240e4)
10. Right click on the DB Connection name -> Disconnect. It is the same step as step 5.
 ![image](https://github.com/dbeaver/dbeaver/assets/1576014/c2f8e001-47c4-4342-b819-9953c364d3e9)
11. **Here is the bug:** The DB Connection name in the UI still has a green checkbox, like it is still connected. The TABLE tab in the middle of the screen also still has active Properties tab. If you try to right-click on the DB Connection name again, the Context Menu will show "Connect" action as available. "Disconnect" action will be grayed out.
 ![image](https://github.com/dbeaver/dbeaver/assets/1576014/902ce49f-1fff-4e4d-82e7-81193b1424d1)
 From this broken state there are plenty of incorrect behavior happens. E.g. trying to open some other table from this connection says "Default context not found". Trying to connect again shows the same error message. Basically the connection is broken and weird at this point and the only way to fix it is to restart DBeaver. "Invalidate/Reconnect" also doesn't help. Sometimes a "Data read" error window pops up with "Can't read data - no active connection" message.
  Trying to refresh the opened table gives this error message: 
![image](https://github.com/dbeaver/dbeaver/assets/1576014/285f619a-18ca-4dda-8d00-17fbb59d20d1)
  Trying to refresh the "Projects" tree removes the green checkbox from the UI, but if you try to connect again, the green checkbox will still be missing. And if you try to open any other table from this connection, a "Default context not found" error will shown again. Trying to access "Data" tab of the table shows this error: 
![image](https://github.com/dbeaver/dbeaver/assets/1576014/2bc28b6c-3c45-41b0-84a3-4ab04b57883b)
Basically, without DBeaver restarting it is impossible to correctly Disconnect. Every time I need to Disconnect some connection I'm forced to restart DBeaver.

### Additional context

This bug never happened before I upgraded to version 24.0.
I tried to install 23.3.5 now and I can't reproduce this bug in this version. So, probably some changes in 24.0.0 caused this bug.

**Comment by ShadelessFox** [comment]
Hello @xak2000,

Can you please reproduce the issue again and provide log files? https://github.com/dbeaver/dbeaver/wiki/Log-files#log-files

**Comment by xak2000** [comment]
Hello @ShadelessFox,

1. Here is the logs just after 2nd disconnet (step 10): [dbeaver-diagnostic-info-1715285599639.zip](https://github.com/dbeaver/dbeaver/files/15267341/dbeaver-diagnostic-info-1715285599639.zip)
2. Here is the logs after I tried to do some actions in this state (open another table, refresh table, invalidate/reconnect, etc.) (step 11): [dbeaver-diagnostic-info-1715285747762.zip](https://github.com/dbeaver/dbeaver/files/15267358/dbeaver-diagnostic-info-1715285747762.zip)

I removed the `configuration.txt` from the archives as this file could contain sensitive information and it's 500KB. I can't analyze the whole 500KB file to understand if it is safe to share. It looks irrelevant to the problem. But if you need it, please ask.

**Comment by E1izabeth** [comment]
I reproduced it
![image](https://github.com/dbeaver/dbeaver/assets/28875055/d5962d36-656b-4401-a433-c28cdd31ed96)


Thank you for bug report.
